bpf/lib/trace_helpers.h

```diff
@@ -0,0 +1,74 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
+/* Copyright Authors of Cilium */
+
+#ifndef TRACE_ID_UTIL_H
+#define TRACE_ID_UTIL_H
+
+#include <bpf/ctx/ctx.h>
+#include <bpf/api.h>
+#include "common.h"
+#include "ip_options.h"
+
+#ifdef IP_TRACING_OPTION_TYPE
+/* Define the ip trace ID map with __u64 trace_id */
+struct {
+	__uint(type, BPF_MAP_TYPE_PERCPU_ARRAY);
+	__uint(max_entries, 1);
+	__type(key, __u32); /* only one key */
+	__type(value, __u64); /* trace_id type */
+	__uint(pinning, LIBBPF_PIN_BY_NAME);
+} trace_id_map __section_maps_btf;
+
+/* bpf_trace_id_set sets the trace_id in the map. */
+static __always_inline __u64 bpf_trace_id_set(__u64 trace_id)
+{
+	__u32 __z = 0;
+	__u64 *__cache = map_lookup_elem(&trace_id_map, &__z);
+
+	if (__cache)
```

Comment:
Question: Under what circumstance would we expect a trace ID to already be there. Is the correct action in such event to overwrite it?

Reply:
Aside from the first ctx/skbuff object processed by the CPU, it will always be 0 or have an ID value.

the value of the per-cpu array is we can store a value for the lifetime of the ctx processed by the CPU (for parsing later when producing dataplane events for control-plane processing.